docs/gc_overview.md

```diff
@@ -23,81 +23,183 @@
 
-To improve performance, allocation caches are reserved in the heap for different threads. These allocation caches are known as thread local heaps (TLH) and allow each thread to allocate memory from its cache without acquiring the heap lock. A TLH is typically used for small objects of less than 512 bytes (768 bytes on 64-bit VMs) although larger objects can be allocated from the cache if there is sufficient space.
+### GC *mark* operation
 
-If a thread allocates a lot of objects, the allocator gives that thread a larger TLH to reduce contention on the heap lock.
+A *mark* operation identifies which objects on the Java heap are reachable from outside of the heap and which objects are unreachable. Reachable objects are in use and must be retained, whereas unreachable objects are not in use and can be cleared away as *garbage*.
 
-A TLH is predefined with an initial default size of 2 KB. On every TLH refresh, the requested size for that thread
-is increased by an increment (default 4 KB). The requested size can grow up to a predefined maximum (default 128 KB).
+The process of marking involves a bit array called a *mark map* that is allocated by the VM at startup, based on the maximum heap size setting. Each bit in the *mark map* corresponds to 8 bytes of heap space. When an object is *marked*, its location in the heap is recorded by setting the appropriate bit in the *mark map*.
 
-After every GC cycle, the TLH requested size for each thread is reduced, sometimes by as much as 50%, to take account of threads that
-reduce their allocation rate and no longer need large TLHs.
+A *mark* operation can be broken down into the following phases:
 
-For very inactive threads, the requested size can even drop below the initial value, down to the predefined minimum (512/768 bytes).
-For very active threads, the maximum TLH requested size might be reached before the next GC occurs.
+1. Initial phase
 
-Larger TLHs can help reduce heap lock contention, but might also reduce heap utilisation and increase heap fragmentation.
+    A root object is an object that is accessible from outside of the heap such as a stack, class static field, or JNI reference. For other objects in the heap to be reachable, they must retain a connection from a root object. In the initial phase, tracing identifies all root objects by running a root scan. Root objects are pushed onto a work stack for processing in the next phase.
 
-The following options control the requested TLH size:
+2. Main phase
 
-- [`-Xgc:tlhMaximumSize=<bytes>`](xgc.md#tlhmaximumsize)
-- [`-Xgc:tlhInitialSize=<bytes>`](xgc.md#tlhinitialsize)
-- [`-Xgc:tlhIncrementSize=<bytes>`](xgc.md#tlhincrementsize)
+    The list of reachable root objects in the work stack is recursively traced for references to other objects in the heap. If new objects are found, they are added to the work stack. If an object is reachable, the appropriate bit is set in the *mark map*.
```

Comment:
Maybe not be obvious to everyone, so you may clarify how we know that _new_ object is found: it's if not marked yet.
So, we will add an object to work stack, only if its bit in mark map is not set (otherwise we would not never complete the process, if there are cycles in the object graph).

Reply:
Updated:

"Objects that are not marked are new objects and these are added to the work stack."